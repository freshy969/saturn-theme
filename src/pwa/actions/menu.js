import * as actionTypes from '../actionTypes';

export const hasOpen = () => ({
  type: actionTypes.MENU_HAS_OPEN,
  event: {
    category: 'Side Menu',
    action: 'open',
  },
});

export const hasClosed = () => ({
  type: actionTypes.MENU_HAS_CLOSED,
});
