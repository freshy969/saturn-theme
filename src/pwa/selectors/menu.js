/* eslint-disable import/prefer-default-export */
export const isOpen = state => state.theme.menu.isOpen;
