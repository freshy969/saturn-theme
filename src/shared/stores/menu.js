import { types } from 'mobx-state-tree';

const Menu = types
  .model('Menu')
  .props({
    isOpen: types.optional(types.boolean, false),
  })
  .actions(self => ({
    hasOpen() {
      if (!self.isOpen) self.isOpen = true;
    },
    hasClosed() {
      if (self.isOpen) self.isOpen = false;
    },
    closeMenuOnRouteChange() {
      
    },
  }));

export default Menu;
