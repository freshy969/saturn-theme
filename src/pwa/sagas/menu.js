import { put, select, takeEvery } from 'redux-saga/effects';
import { dep } from 'worona-deps';
import * as actions from '../actions';
import * as selectors from '../selectors';

function* closeMenuOnRouteChange() {
  const isOpen = yield select(state => selectors.menu.isOpen(state));

  if (isOpen) yield put(actions.menu.hasClosed());
}

export default function* menuSagas() {
  yield takeEvery(
    dep('connection', 'actionTypes', 'ROUTE_CHANGE_REQUESTED'),
    closeMenuOnRouteChange,
  );
}
