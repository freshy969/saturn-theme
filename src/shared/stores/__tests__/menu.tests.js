import menu from '../menu';

describe('Theme › Shared › Stores › Menu', () => {
  test('store has isOpen prop', () => {
    const self = menu.create();

    expect(self.isOpen).toBe(false);
  });

  test('open() changes value of isOpen to true', () => {
    const self = menu.create();

    Object.defineProperty(self, 'open', {
      writable: true,
      value: jest.fn(self.open),
    });

    expect(self.isOpen).toBe(false);

    self.open();

    expect(self.open).toHaveBeenCalled();
    expect(self.isOpen).toBe(true);
  });

  test('close() changes value of isOpen to false', () => {
    const self = menu.create({ isOpen: true });

    Object.defineProperty(self, 'close', {
      writable: true,
      value: jest.fn(self.close),
    });

    expect(self.isOpen).toBe(true);

    self.close();

    expect(self.close).toHaveBeenCalled();
    expect(self.isOpen).toBe(false);
  });
});
