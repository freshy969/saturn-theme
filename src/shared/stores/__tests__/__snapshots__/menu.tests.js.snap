// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme › Shared › Stores › Menu store has isOpen prop 1`] = `
Object {
  "isOpen": false,
}
`;
